ACTIN-1834: Change PARTIAL_LOSS for clarity

zoehehartwig opened the pull request.

zoehehartwig commented:
I did all the changes. The only references to loss in the code are either related to heterozygosity or the one reference to HLA-1 loss of function. There are some references to loss in the README that I wasn't sure about.

pauldwolfe commented on orange/README.md:
```diff
@@ -141,10 +141,10 @@ interesting and added to the report:
     1. Gains in genes for which we report amplifications with a relative minimum copy number between 2.5 and 3 times ploidy.
-    2. Any chromosomal band location with at least one gene lost or fully amplified is considered potentially interesting.
-       A maximum of 10 additional gains (sorted by minimum copy number) and 10 additional losses are reported as potentially interesting:
+    2. Any chromosomal band location with at least one gene deletedc or fully amplified is considered potentially interesting.
```
spelling

pauldwolfe commented on orange/src/main/java/com/hartwig/hmftools/orange/algo/purple/CopyNumberSelector.java:
```diff
@@ -143,59 +143,59 @@ private static List<PurpleGainLoss> selectInterestingGains(@NotNull List<PurpleG
 
-        List<PurpleGainLoss> lossesAutosomes = Lists.newArrayList();
-        for(PurpleGainLoss loss : unreportedLosses)
+        List<PurpleGainDeletion> DelsAutosomes = Lists.newArrayList();
```
capitalized D should be lowercase.

pauldwolfe reviewed (approved):
Couple small comments to address but otherwise looks good. 

As discussed, adding backwards compatibility for this change would be quite some effort so we'll make this a breaking change and a new major orange version.

kduyvesteyn commented on orange/src/test/java/com/hartwig/hmftools/orange/algo/util/GermlineConversionTest.java:
```diff
@@ -106,9 +106,9 @@ public void canConvertPurple()
-        PurpleGainLoss reportableGermlineFullLoss = TestPurpleGainLossFactory.builder().build();
+        PurpleGainDeletion somaticGainDel = TestPurpleGainDeletionFactory.builder().build();
+        PurpleGainDeletion reportableSomaticGainDel = TestPurpleGainDeletionFactory.builder().build();
+        PurpleGainDeletion reportableGermlineFullLoss = TestPurpleGainDeletionFactory.builder().build();
```
One missed rename here

kduyvesteyn commented on orange/src/main/java/com/hartwig/hmftools/orange/report/tables/ImmuneEscapeTable.java:
```diff
@@ -21,7 +21,7 @@ public static Table build(@NotNull String title, float width, @NotNull ImmuneEsc
                 new Cell[] { cells.createHeader("Escape Mechanism"), cells.createHeader("Detected?"), cells.createHeader(Strings.EMPTY) });
 
-        table.addCell(cells.createContent("HLA-1 loss-of-function"));
+        table.addCell(cells.createContent("HLA-1 =loss-of-function"));
```
Fat finger?

kduyvesteyn commented on orange/src/main/java/com/hartwig/hmftools/orange/report/chapters/SomaticFindingsChapter.java:
```diff
@@ -156,10 +156,10 @@ private void addKataegisPlot(@NotNull Document document)
-        String nearDriverGainsTitle = "Potentially interesting near-driver amps";
-        String suspectGainsTitle = "Other regions with amps";
-        String suspectLossesTitle = "Regions with deletions in genes in other autosomal regions";
+        String driverAmpsDelsTitle = "Driver amplifications/homologous deletions";
```
It is a "homozygous" deletion, not a "homologous" deletion! 

Also I would just use " and" rather than "/"

kduyvesteyn commented on orange/src/test/java/com/hartwig/hmftools/orange/algo/purple/PurpleInterpreterTest.java:
```diff
@@ -364,14 +364,13 @@ private static PurpleInterpreter createInterpreter(@NotNull EnsemblDataCache ens
         return new PurpleInterpreter(
-                purpleVariantFactory,
-                germlineGainLossFactory,
+                purpleVariantFactory, germlineGainDeletionFactory,
```
Formatting could be improved

kduyvesteyn commented on orange/src/test/java/com/hartwig/hmftools/orange/algo/util/GermlineConversionTest.java:
```diff
@@ -359,9 +359,9 @@ public void mergesSomaticAndGermlineFullLosses()
                 .gene(TEST_GENE1)
                 .build();
-        PurpleGainLoss somaticLoss = TestPurpleGainLossFactory.builder()
+        PurpleGainDeletion somaticLoss = TestPurpleGainDeletionFactory.builder()
```
One rename missed here.

kduyvesteyn commented on orange/src/test/java/com/hartwig/hmftools/orange/algo/util/GermlineConversionTest.java:
```diff
@@ -372,9 +372,9 @@ public void mergesSomaticAndGermlineFullLosses()
                 .gene(TEST_GENE1)
                 .build();
-        PurpleGainLoss germlineLoss = TestPurpleGainLossFactory.builder()
+        PurpleGainDeletion germlineLoss = TestPurpleGainDeletionFactory.builder()
```
One rename missed here

kduyvesteyn commented on orange/src/test/java/com/hartwig/hmftools/orange/algo/util/GermlineConversionTest.java:
```diff
@@ -407,9 +407,9 @@ public void mergesSomaticPartialAndGermlineFullLoss()
                 .gene(TEST_GENE1)
                 .build();
-        PurpleGainLoss somaticLoss = TestPurpleGainLossFactory.builder()
+        PurpleGainDeletion somaticLoss = TestPurpleGainDeletionFactory.builder()
```
One rename missing here

kduyvesteyn commented on orange/src/test/java/com/hartwig/hmftools/orange/algo/util/GermlineConversionTest.java:
```diff
@@ -420,9 +420,9 @@ public void mergesSomaticPartialAndGermlineFullLoss()
                 .gene(TEST_GENE1)
                 .build();
-        PurpleGainLoss germlineLoss = TestPurpleGainLossFactory.builder()
+        PurpleGainDeletion germlineLoss = TestPurpleGainDeletionFactory.builder()
```
One missing rename here

kduyvesteyn commented on orange/src/test/java/com/hartwig/hmftools/orange/algo/util/GermlineConversionTest.java:
```diff
@@ -455,9 +455,9 @@ public void mergesSomaticFullAndGermlinePartialLoss()
                 .gene(TEST_GENE1)
                 .build();
-        PurpleGainLoss somaticLoss = TestPurpleGainLossFactory.builder()
+        PurpleGainDeletion somaticLoss = TestPurpleGainDeletionFactory.builder()
```
One missed rename here

kduyvesteyn commented on orange/src/test/java/com/hartwig/hmftools/orange/algo/util/GermlineConversionTest.java:
In general some renames missed in this file. Please review!

kduyvesteyn reviewed (commented):
Good work @zoehehartwig !

All my comments are fairly minor, though one thing that is odd that needs some more investigation: You haven't updated the actual datamodels in minimally.empty.orange.json and real.orange.json yet the unit tests succeed. How is that possible? That is not supposed to happen. Maybe something you can look into with @pauldwolfe ?

zoehehartwig commented:
I changed all occurances of "loss" to "del" in the JSONs. As for why they worked before, I think it's because it doesn't matter what the variable name is in the JSON. When I replaced the variable names with a random set of letters, it still worked.

pauldwolfe commented:
@zoehehartwig @kduyvesteyn Just looking into the question above about the test JSON files. 

Indeed it seems that Gson deserialization will default those collections to empty list if they are not present in the json (so as @zoehehartwig says, you can name them anything and you end up with empty lists).  As both the tests for the "real.orange.json" and "empty.orange.json" only check that the file can be deserialized and not specific values, the tests pass.

Note that the test for "minimally.populated" will fail if the fields aren't updated. 

I've been looking for the magic option to tell Gson (or Immutables) to be more strict but haven't found it yet. @zoehehartwig could you follow up on that?
